source/core/ib_particle.cc

```diff
@@ -101,54 +101,44 @@ template <int dim>
-  properties[PropertiesIndex::vz] = std::make_pair("Velocity", 1);
-  properties[PropertiesIndex::fx] = std::make_pair("Force", 3);
-  properties[PropertiesIndex::fy] = std::make_pair("Force", 1);
-  properties[PropertiesIndex::fz] = std::make_pair("Force", 1);
-  properties[PropertiesIndex::m]  = std::make_pair("Mass", 1);
-  properties[PropertiesIndex::ox] = std::make_pair("Omega", 3);
-  properties[PropertiesIndex::oy] = std::make_pair("Omega", 1);
-  properties[PropertiesIndex::oz] = std::make_pair("Omega", 1);
-  properties[PropertiesIndex::tx] = std::make_pair("Torques", 3);
-  properties[PropertiesIndex::ty] = std::make_pair("Torques", 1);
-  properties[PropertiesIndex::tz] = std::make_pair("Torques", 1);
+  std::vector<std::pair<std::string, int>> particle_properties(
+    DEM::PropertiesIndex::n_properties);
+  particle_properties = DEM::DEMProperties<dim>::get_properties_name();
 
-  return properties;
+  // Rename particle type to id to match IB pattern
+  particle_properties[DEM::PropertiesIndex::type] = std::make_pair("id", 1);
+
+  return particle_properties;
 }
 
 template <int dim>
 std::vector<double>
 IBParticle<dim>::get_properties()
 {
   std::vector<double> properties(get_number_properties());
-  properties[0]  = particle_id;
-  properties[1]  = radius * 2.0;
-  properties[2]  = velocity[0];
-  properties[3]  = velocity[1];
-  properties[5]  = fluid_forces[0];
-  properties[6]  = fluid_forces[1];
-  properties[8]  = omega[0];
-  properties[9]  = omega[1];
-  properties[10] = omega[2];
-  properties[11] = mass;
-  properties[12] = fluid_torque[0];
-  properties[13] = fluid_torque[1];
-  properties[14] = fluid_torque[2];
+  properties[DEM::PropertiesIndex::type]         = particle_id;
+  properties[DEM::PropertiesIndex::dp]           = radius * 2.0;
+  properties[DEM::PropertiesIndex::v_x]          = velocity[0];
+  properties[DEM::PropertiesIndex::v_y]          = velocity[1];
+  properties[DEM::PropertiesIndex::omega_x]      = omega[0];
+  properties[DEM::PropertiesIndex::omega_y]      = omega[1];
+  properties[DEM::PropertiesIndex::omega_z]      = omega[2];
+  properties[DEM::PropertiesIndex::fem_force_x]  = fluid_forces[0];
+  properties[DEM::PropertiesIndex::fem_force_y]  = fluid_forces[1];
+  properties[DEM::PropertiesIndex::fem_torque_x] = fluid_torque[0];
+  properties[DEM::PropertiesIndex::fem_torque_y] = fluid_torque[1];
+  properties[DEM::PropertiesIndex::fem_torque_z] = fluid_torque[2];
+  properties[DEM::PropertiesIndex::mass]         = mass;
+
   if (dim == 2)
     {
-      properties[4] = 0;
-      properties[7] = 0;
+      properties[DEM::PropertiesIndex::v_z]         = 0.;
+      properties[DEM::PropertiesIndex::fem_force_z] = 0.;
     }
   if (dim == 3)
     {
-      properties[4] = velocity[2];
-      properties[7] = fluid_forces[2];
+      properties[DEM::PropertiesIndex::v_z]         = velocity[2];
+      properties[DEM::PropertiesIndex::fem_force_z] = fluid_forces[2];
     }
 
   return properties;
```

Comment:
This seems good to me! This will help a lot with the maintenance of the code in the long term.